sdk/src/main/scala/com/horizen/account/state/AccountState.scala

```diff
@@ -347,20 +350,31 @@ class AccountState(val params: NetworkParams,
 
   override def getCode(address: Array[Byte]): Array[Byte] = using(getView)(_.getCode(address))
 
-  override def getBaseFee: BigInteger = using(getView)(_.getBaseFee)
-
-  override def getBlockGasLimit: BigInteger = using(getView)(_.getBlockGasLimit)
-
   override def validate(tx: SidechainTypes#SCAT): Try[Unit] = Try {
     tx.semanticValidity()
 
     if (tx.isInstanceOf[EthereumTransaction]) {
 
       val ethTx = tx.asInstanceOf[EthereumTransaction]
-      val blockGasPool = new GasPool(getBlockGasLimit)
+      val blockGasLimit = Account.GAS_LIMIT
+      val blockGasPool = new GasPool(BigInteger.valueOf(blockGasLimit))
+      val blockContext = new BlockContext(
+        // use the null address as forger
+        new Array[Byte](32),
+        // TODO: what timestamp do we use here? just use "now()" or can we get the timestamp of the last block?
+        0,
+        // TODO: what baseFee do we use here? can we get the baseFee of the last block?
+        0,
+        // TODO: can we get the gas limit of the last block?
+        blockGasLimit,
+        stateMetadataStorage.getHeight,
+        // TODO: can the consensus epoch number be None here?
+        stateMetadataStorage.getConsensusEpochNumber.get,
+        stateMetadataStorage.getWithdrawalEpochInfo.epoch
+      )
```

Comment:
I could use some feedback regarding the TODOs here.

Reply:
1-3.) If we pass the block here instead of or in addition to tx we can surely do that, because you then get the parent block id, too.
Do we want to do that @i-Alex?

Reply:
4.) afaik no at this point